src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,351 @@
+                        tasks.add(event);
+
+                        if (savedTask[1].equals("X")) {
+                            tasks.get(counter).setComplete();
+                        }
+                        break;
+                    }
+                }
+                counter++;
+            }
+
+        } catch (IOException e) {
+            System.out.println("An error has occurred. Please try again.");
+            System.exit(1);
+        }
+    }
+
+    public static void endFileProcessing(ArrayList<Task> tasks) {
+        try {
+            System.out.println("Saving current tasks...");
+            FileWriter saveFile = new FileWriter("duke.txt");
+
+            for (int i = 0; i < tasks.size(); ++i) {
+                String task = String.valueOf(tasks.get(i));
+                String[] splitTaskDescription = task.split("] ", 3);
+                saveFile.write(tasks.get(i).getType() + " / "  + tasks.get(i).getStatus() + " / " + splitTaskDescription[2] + '\n');
+            }
+
+            saveFile.close();
+        } catch (IOException e) {
+            System.out.println("An error has occurred. Please try again.");
+            System.exit(1);
+        }
+    }
+
+    /**
+     * Prints an acknowledgement, based on the type of task.
+     * Also prints the description of the task and the current number of tasks
+     */
+    public static void printAcknowledgement(String taskType, String description, String size) {
+        printHorizontalRule();
+        System.out.println("Added " + taskType + ": " + description);
+        System.out.println("Total Number of Tasks: " + size);
+        printHorizontalRule();
+    }
+
+    /**
+     * Removes the command word for creating tasks, ie. to do, deadline or event
+     */
+    public static String removeCommandWord(String userInput) {
+        return userInput.split(" ", 2)[1];
+    }
+
+    /**
+     * Extracts the description given by the user for deadline tasks
+     */
+    public static String getDescriptionForDeadline(String input) {
+        return input.split(" /by ", 2)[0];
+    }
+
+    /**
+     * Extracts the deadline for deadline tasks
+     */
+    public static String getDeadline(String input) {
+        return input.split(" /by ", 2)[1];
+    }
+
+    /**
+     * Extracts the description given by the user for event tasks
+     */
+    public static String getDescriptionForEvent(String input) {
+        return input.split(" /from ", 2)[0];
+    }
+
+    /**
+     * Extracts the start and end timing for event tasks
+     */
+    public static String[] getTimings(String input) {
+
+        String[] timings = new String[2];
+        String timing = input.split(" /from ", 2)[1];
+
+        timings[0] = timing.split(" /to ", 2)[0];
+        timings[1] = timing.split(" /to ", 2)[1];
+
+        return timings;
+    }
+
+
+    /* --- MAIN FUNCTION --- */
+    public static void main(String[] args) {
```

Comment:
This is particularly long method.
Many operations can be extracted out as separate methods.